core/src/main/scala/org/http4s/parser/QueryParser.scala

```diff
@@ -91,7 +90,7 @@ private[http4s] class QueryParser(
     // begin iterating through the chars
     while (error == null && input.hasRemaining) {
       val c = input.get()
-      (c: @switch) match {
+      c match {
```

Comment:
dotty doesn't like switch statements with guards, and if I remove the guards by hand we have too few conditions, dotty and scala only compile to a switch with more than 4 cases